services/wallet/router_send_type.go

```diff
@@ -0,0 +1,169 @@
+	walletCommon "github.com/status-im/status-go/services/wallet/common"
+	"github.com/status-im/status-go/services/wallet/token"
+	"github.com/status-im/status-go/transactions"
+)
+
+type SendType int
+
+const (
+	Transfer SendType = iota
+	ENSRegister
+	ENSRelease
+	ENSSetPubKey
+	StickersBuy
+	Bridge
+	ERC721Transfer
+	ERC1155Transfer
+	Swap
+)
+
+func (s SendType) IsCollectiblesTransfer() bool {
+	return s == ERC721Transfer || s == ERC1155Transfer
+}
+
+func (s SendType) FetchPrices(service *Service, tokenID string) (map[string]float64, error) {
+	symbols := []string{tokenID, "ETH"}
+	if s.IsCollectiblesTransfer() {
+		symbols = []string{"ETH"}
+	}
+
+	pricesMap, err := service.marketManager.FetchPrices(symbols, []string{"USD"})
+	if err != nil {
+		return nil, err
+	}
+	prices := make(map[string]float64, 0)
+	for symbol, pricePerCurrency := range pricesMap {
+		prices[symbol] = pricePerCurrency["USD"]
+	}
+	if s.IsCollectiblesTransfer() {
+		prices[tokenID] = 0
+	}
+	return prices, nil
+}
+
+func (s SendType) FindToken(service *Service, account common.Address, network *params.Network, tokenID string) *token.Token {
+	if !s.IsCollectiblesTransfer() {
+		return service.tokenManager.FindToken(network, tokenID)
+	}
+
+	parts := strings.Split(tokenID, ":")
+	contractAddress := common.HexToAddress(parts[0])
+	collectibleTokenID, success := new(big.Int).SetString(parts[1], 10)
+	if !success {
+		return nil
+	}
+	uniqueID, err := service.collectibles.GetOwnedCollectible(walletCommon.ChainID(network.ChainID), account, contractAddress, collectibleTokenID)
+	if err != nil || uniqueID == nil {
+		return nil
+	}
+
+	return &token.Token{
+		Address:  contractAddress,
+		Symbol:   collectibleTokenID.String(),
+		Decimals: 0,
+		ChainID:  network.ChainID,
+	}
+}
+
+func (s SendType) isTransfer() bool {
+	return s == Transfer || s == Swap || s.IsCollectiblesTransfer()
+}
+
+func (s SendType) needL1Fee() bool {
+	return s != ENSRegister && s != ENSRelease && s != ENSSetPubKey && s != StickersBuy
+}
+
+func (s SendType) canUseBridge(b bridge.Bridge) bool {
+	if s == ERC721Transfer && b.Name() != ERC721TransferString {
+		return false
+	}
+
+	if s != ERC721Transfer && b.Name() == ERC721TransferString {
+		return false
+	}
+
+	if s == ERC1155Transfer && b.Name() != ERC1155TransferString {
+		return false
+	}
+
+	if s != ERC1155Transfer && b.Name() == ERC1155TransferString {
+		return false
+	}
+
+	return true
+}
+
+func (s SendType) isAvailableFor(network *params.Network) bool {
+	if s == Swap {
+		return network.ChainID == walletCommon.EthereumMainnet ||
+			network.ChainID == walletCommon.OptimismMainnet ||
+			network.ChainID == walletCommon.ArbitrumMainnet
+	}
+
+	if s == Transfer || s == Bridge || s.IsCollectiblesTransfer() {
+		return true
+	}
+
+	if network.ChainID == walletCommon.EthereumMainnet ||
+		network.ChainID == walletCommon.EthereumGoerli ||
+		network.ChainID == walletCommon.EthereumSepolia {
+		return true
+	}
+
+	return false
+}
```

Comment:
This function has a number of branching conditionals, so perhaps we could declare the conditions first and then compare after. Something like:

```go
func (s SendType) isAvailableFor(network *params.Network) bool {
    // Set of network ChainIDs allowed for any type of transaction
    allAllowedNetworks := map[uint64]bool{
        walletCommon.EthereumMainnet: true,
        walletCommon.EthereumGoerli:  true,
        walletCommon.EthereumSepolia: true,
    }

    // Additional specific networks for the Swap SendType
    swapAllowedNetworks := map[uint64]bool{
        walletCommon.EthereumMainnet:  true,
        walletCommon.OptimismMainnet:  true,
        walletCommon.ArbitrumMainnet:  true,
    }

    // Check for Swap specific networks
    if s == Swap {
        return swapAllowedNetworks[network.ChainID]
    }

    // Check for any SendType available for all networks
    if s == Transfer || s == Bridge || s.IsCollectiblesTransfer() || allAllowedNetworks[network.ChainID] {
        return true
    }

    return false
}
```

This approach makes it more clear what the networks are used for what purpose.